Kitodo-API/src/main/java/org/kitodo/api/filemanagement/FileManagementInterface.java

```diff
@@ -38,24 +40,89 @@
      */
     URI create(URI parentFolderUri, String name, boolean file) throws IOException;
 
+    /**
+     * Attempts to get locks on one or more files. There are only two results:
+     * either, all locks can be granted, or none of the requested locks are
+     * granted. In the former case, the conflicts map in the locking result is
+     * empty, in the latter case it contains for each conflicting file the users
+     * who hold a conflicting lock on the file. If no locks have been granted,
+     * the call to {@code close()} on the locking result is meaningless, meaning
+     * that leaving the try-with-resources statement will not throw an
+     * exception. Just to mention that.
+     * 
+     * @param user
+     *            A human-readable string that identifies the user or process
+     *            requesting the locks. This string will later be returned to
+     *            other users if they try to request a conflicting lock.
+     * @param requests
+     *            the locks to request
+     * @return An object that manages allocated locks or provides information
+     *         about conflict originators in case of error.
+     * @throws IOException
+     *             if the file does not exist or if an error occurs in disk
+     *             access, e.g. because the write permission for the directory
+     *             is missing
+     */
+    LockingResult tryLock(String user, Map<URI, LockingMode> requests) throws IOException;
+
     /**
      * Opens an OutputStream to a given uri.
      *
      * @param uri
      *            the uri to write to
      * @return an writable OutputStream
+     * @deprecated This write function creates an exclusive lock with the
+     *             meaningless user name “System”. Therefore, this writing
+     *             function should not be used anymore. Use
+     *             {@link #write(URI, LockingResult)} instead.
      */
+    @Deprecated
     OutputStream write(URI uri) throws IOException;
 
+    /**
+     * Opens an OutputStream to a given uri.
+     *
+     * @param uri
+     *            the uri to write to
+     * @param permission
+     *            the result of a successful lock operation that authorizes the
+     *            opening of the stream
+     * @return an writable OutputStream
+     * @throws ProtocolException
+     *             if the file had to be first read in again, but this step was
+     *             skipped on the protocol. This error can occur with the
+     *             UPGRADE_WRITE_ONCE lock because its protocol form requires
+     *             that the file must first be read in again and the input
+     *             stream must be closed after the lock has been upgraded.
+     */
+    OutputStream write(URI uri, LockingResult permission) throws IOException;
```

Comment:
Yes, the method throws an I/O excepiton. That's not explained, because that's obvious when opening an output stream. In the existing methods, the I/O exceptions are nowhere explained. I agree with that. What I have explained are the exceptions that have a special meaning. Protocol exception is a subclass of I/O exception and therefore does not need to be declared explicitly.